app/src/main/java/com/farmerbb/taskbar/content/TaskbarPosition.java

```diff
@@ -0,0 +1,198 @@
+                    case Surface.ROTATION_270:
+                        return POSITION_BOTTOM_VERTICAL_LEFT;
+                }
+                break;
+            case POSITION_BOTTOM_VERTICAL_RIGHT:
+                switch (rotation) {
+                    case Surface.ROTATION_0:
+                        return POSITION_BOTTOM_VERTICAL_RIGHT;
+                    case Surface.ROTATION_90:
+                        return POSITION_TOP_RIGHT;
+                    case Surface.ROTATION_180:
+                        return POSITION_TOP_VERTICAL_LEFT;
+                    case Surface.ROTATION_270:
+                        return POSITION_BOTTOM_LEFT;
+                }
+                break;
+            case POSITION_TOP_LEFT:
+                switch (rotation) {
+                    case Surface.ROTATION_0:
+                        return POSITION_TOP_LEFT;
+                    case Surface.ROTATION_90:
+                        return POSITION_BOTTOM_VERTICAL_LEFT;
+                    case Surface.ROTATION_180:
+                        return POSITION_BOTTOM_RIGHT;
+                    case Surface.ROTATION_270:
+                        return POSITION_TOP_VERTICAL_RIGHT;
+                }
+                break;
+            case POSITION_TOP_VERTICAL_LEFT:
+                switch (rotation) {
+                    case Surface.ROTATION_0:
+                        return POSITION_TOP_VERTICAL_LEFT;
+                    case Surface.ROTATION_90:
+                        return POSITION_BOTTOM_LEFT;
+                    case Surface.ROTATION_180:
+                        return POSITION_BOTTOM_VERTICAL_RIGHT;
+                    case Surface.ROTATION_270:
+                        return POSITION_TOP_RIGHT;
+                }
+                break;
+            case POSITION_TOP_RIGHT:
+                switch (rotation) {
+                    case Surface.ROTATION_0:
+                        return POSITION_TOP_RIGHT;
+                    case Surface.ROTATION_90:
+                        return POSITION_TOP_VERTICAL_LEFT;
+                    case Surface.ROTATION_180:
+                        return POSITION_BOTTOM_LEFT;
+                    case Surface.ROTATION_270:
+                        return POSITION_BOTTOM_VERTICAL_RIGHT;
+                }
+                break;
+            case POSITION_TOP_VERTICAL_RIGHT:
+                switch (rotation) {
+                    case Surface.ROTATION_0:
+                        return POSITION_TOP_VERTICAL_RIGHT;
+                    case Surface.ROTATION_90:
+                        return POSITION_TOP_LEFT;
+                    case Surface.ROTATION_180:
+                        return POSITION_BOTTOM_VERTICAL_LEFT;
+                    case Surface.ROTATION_270:
+                        return POSITION_BOTTOM_RIGHT;
+                }
+                break;
+        }
+        return POSITION_BOTTOM_LEFT;
+    }
+
+    public static boolean isVertical(String position) {
```

Comment:
With the `isVertical()`, etc.  methods, as far as I can tell, the argument is always going to be the result of `U.getTaskbarPosition(context)`.  In order to simplify things outside of this class, could we overload these methods to pass in a `Context` object instead, that will call `U.getTaskbarPosition(context)` for us and then pass the result on as a String argument?  We can then leave the versions with String as a parameter for testing purposes.

Reply:
Okay, it's simpler. I will add a new commit to do it in this `PR`. Thanks your advice.